docs/ref/templates/language.txt

```diff
@@ -434,6 +434,67 @@ wouldn't know which one of the blocks' content to use.
 .. _next section: #automatic-html-escaping
 .. _automatic-html-escaping:
 
+Circular Template inheritance
+-----------------------------
+
+Also known as recursive template inheritance, or over-extension, this is a very
+powerful feature of Django's template inheritance. This allows many templates
+to add incremental additions to each other rather than needing to always
+override outright.
+
+The primary use-case for this type of inheritance is to simultaneously override
+and extend templates from other reusable apps, in your own Django project.
+
+Consider the following project:
+
+.. snippet::
+    :filename: settings.py
+
+    INSTALLED_APPS = (
+        "app1",
+        "app2",
+        "app3",
```

Comment:
I think these need to be the reversed (or the templates adjusted). As it stands `app1/templates/page.html` will be found first. This extends nothing so will just be rendered as is.